src/main/scala/org/bblfsh/client/v2/BblfshClient.scala

```diff
@@ -0,0 +1,161 @@
+package org.bblfsh.client.v2
+
+import java.nio.ByteBuffer
+
+import com.google.protobuf.ByteString
+import com.google.protobuf.duration.Duration
+import io.grpc.ManagedChannelBuilder
+import gopkg.in.bblfsh.sdk.v2.protocol.driver.{ParseRequest, ParseResponse, DriverGrpc, SupportedLanguagesRequest, SupportedLanguagesResponse, VersionRequest, VersionResponse, DriverHostGrpc}
+
+import org.bblfsh.client.v2.libuast.Libuast
+
+
+class BblfshClient(host: String, port: Int, maxMsgSize: Int) {
+  private val DEFAULT_TIMEOUT_SEC = 5
+
+  private val channel = ManagedChannelBuilder
+    .forAddress(host, port)
+    .usePlaintext(true)
+    .maxInboundMessageSize(maxMsgSize)
+    .build()
+
+  private val stub = DriverGrpc.blockingStub(channel)
+  private val stubInfo = DriverHostGrpc.blockingStub(channel)
+
+  /**
+    * Parses file with a given name and content using 
+    * the provided timeout.
+    *
+    * @param name file name
+    * @param content file content
+    * @param lang (optional) language to parse, default auto-detect \w enry
+    * @param timeout (disabled) bblfsh request timeout, seconds
+    *                Right now this does not have any effect in v2.
+    * @return UAST in parse response.
+    */
+  def parseWithOptions(
+    name: String,
+    content: String,
+    lang: String,
+    timeout: Long
+  ): ParseResponse = {
+    // FIXME(bzz): make timout work in v2 again
+    val req = ParseRequest(
+      filename = name,
+      content = content,
+      language = lang
+    )
+    stub.parse(req)
+  }
+
+  /**
+    * Parses the given file name and content.
+    *
+    * A default timeout of 5 seconds will be applied, same as was done by the server
+    * before https://github.com/bblfsh/bblfshd/blob/83166ea0087bfe20c24fc471309f70f422383198/daemon/pool.go#L191
+    *
+    * @param name file name
+    * @param content file content
+    * @param lang (optional) language to parse, default auto-detect \w enry
+    * @return UAST in parse response.
+    */
+  def parse(
+    name: String,
+    content: String,
+    lang: String = ""
+  ): ParseResponse = parseWithOptions(name, content, lang, DEFAULT_TIMEOUT_SEC)
```

Comment:
Maybe we can just add another optional parameter for the timeout? It shouldn't break the caller, right?

Reply:
Not sure what you mean, for now it just keeps the same API as it was before.

Reply:
I mean that there is a `parseWithOptions` function that seems unnecessary. If I understood correctly, the `parse` function may include the `timeout` argument with a default value, so there is only one call instead of two, but it's still backward-compatible because the parameter is optional. I'm probably missing something here.

Reply:
Thanks! Added to timeout todo item in the umbrella issue - if you do not mind, I would prefer to apply sam logic as in https://github.com/bblfsh/client-scala/pull/84#discussion_r288460731 applied